core/internal/filetransfer/file_transfer_azure.go

```diff
@@ -124,27 +135,41 @@ type AzureFileTransfer struct {
 
 	// blobClient is a client for a specific blob
 	blobClient AzureBlobClient
+
+	// blockBlobClient is a client for a specific blob
+	blockBlobClient AzureBlockBlobClient
+}
+
+type AzureClientOverrides struct {
```

Comment:
created this type so that we don't have so many parameters to `NewAzureFileTransfer`, and to match the other file transfers that just have one parameter for the override client